DEPRECATIONS.md

```diff
@@ -0,0 +1,21 @@
+# Deprecations
+
+This document lists deprecated constructs in the SDK, with their expected time (⌛) of further deprecations and removals.
+
+| API / Feature | Deprecated (warning) | Deprecated (error) | Removed | Notes |
+| --- | --- | --- | --- | --- |
+| `setUser` (and similar) methods<br/>*client* | 2021.02.03.<br/>4.5.3 | 2021.05.03.⌛ | 2021.08.03.⌛ | Replaced by `connectUser` style methods that return `Call` objects, see migration guide (TODO) |
+| `MessageListViewModel.Event.AttachmentDownload`<br/>*ui-common* | 2021.01.29.<br/>4.5.2 | 2021.02.29.⌛ | 2021.03.29.⌛ | Use `DownloadAttachment` instead |
+| `subscribe` methods with Kotlin function parameters<br/>*client* | 2021.01.27.<br/>4.5.2 | 2021.03.27.⌛ | 2021.05.27.⌛ | Use methods with `ChatEventListener` parameters instead (only affects Java clients) |
```

Comment:
`2021.01.27.` Do we need dots in the end of every date?

Reply:
Probably not, I'll remove them 😄

Reply:
What about to use `/` or `-` as date separator? :thinking:

Reply:
![image](https://user-images.githubusercontent.com/12054216/108516802-f1c14b80-72c6-11eb-905f-68dbb13a7395.png)

Made me write a regex, but okay 😄

Reply:
![image](https://user-images.githubusercontent.com/12054216/108516890-0998cf80-72c7-11eb-914e-96811f56d7d5.png)

Not okay, breaks GitHub formatting

Reply:
I don't want to add even more manual `<br/>`s into the doc, editing it is already gonna be painful